File: final/pwm.c
#include <wiringPi.h>
#include <stdio.h>
#include <stdlib.h>
#include <stdint.h>

// GPIO17 = GPIO_GEN0 = P1.11, connecting a red led
#define PIN         0
#define PWM_RANGE   100

int main (void)
{
    int brightness ;
    printf ("Raspberry Pi wiringPi PWM test program\n") ;
    if (wiringPiSetup () == -1)
        exit (1) ;
    pinMode (PIN, PWM_OUTPUT);
    softPwmCreate(PIN, 0, PWM_RANGE);

    for (;;) {
        for (brightness = 0 ; brightness < PWM_RANGE ; ++brightness) {
//            pwmWrite (PIN, brightness) ;
            softPwmWrite(PIN, brightness);
            delay (10) ;
        }
        for (brightness = PWM_RANGE-1 ; brightness >= 0 ; --brightness) {
//            pwmWrite (PIN, brightness) ;
            softPwmWrite(PIN, brightness);
            delay (10) ;
        }
    }
    return 0 ;
}

File: final/pwm.py
# softPwm uses a fixed frequency
import wiringpi2

OUTPUT = 1

# GPIO17 = GPIO_GEN0 = P1.11, connecting a red led
PIN_TO_PWM = 0

wiringpi2.wiringPiSetup()
wiringpi2.pinMode(PIN_TO_PWM,OUTPUT)
wiringpi2.softPwmCreate(PIN_TO_PWM,0,100) # Setup PWM using Pin, Initial Value and Range parameters

for time in range(0,4):
        for brightness in range(0,100): # Going from 0 to 100 will give us full off to full on
                wiringpi2.softPwmWrite(PIN_TO_PWM,brightness) # Change PWM duty cycle
                wiringpi2.delay(10) # Delay for 0.2 seconds
        for brightness in reversed(range(0,100)):
                wiringpi2.softPwmWrite(PIN_TO_PWM,brightness)
                wiringpi2.delay(10)
